x-pack/legacy/plugins/canvas/public/lib/workpad_service.js

```diff
@@ -12,6 +12,35 @@ import {
 } from '../../common/lib/constants';
 import { fetch } from '../../common/lib/fetch';
 import { getCoreStart } from '../legacy';
+/*
+  Remove any top level keys from the workpad which will be rejected by validation
+*/
+const validKeys = [
```

Comment:
Ideally this should come from the WorkpadSchema that is in the new plugin, but the kbn-config-schema uses JOI under the hood, so I don't think pulling it into the browser is ideal?